framework/math/SpatialDiscretization/FiniteElement/Lagrange/LagrangeBase.cc

```diff
@@ -0,0 +1,149 @@
+#include "LagrangeBase.h"
+
+#include "mesh/MeshContinuum/chi_meshcontinuum.h"
+
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeSlabMapping.h"
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeQuadMapping.h"
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeTriangleMapping.h"
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeHexMapping.h"
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeWedgeMapping.h"
+#include "math/SpatialDiscretization/CellMappings/Lagrange/LagrangeTetMapping.h"
+
+#include "math/SpatialDiscretization/CellMappings/PieceWiseLinear/PieceWiseLinearPolygonMapping.h"
+#include "math/SpatialDiscretization/CellMappings/PieceWiseLinear/PieceWiseLinearPolyhedronMapping.h"
+
+#include "chi_log_exceptions.h"
+
+namespace chi_math::spatial_discretization
+{
+
+LagrangeBase::LagrangeBase(const chi_mesh::MeshContinuum& grid,
+                           QuadratureOrder q_order,
+                           SDMType sdm_type,
+                           CoordinateSystemType cs_type)
+  : FiniteElementBase(grid, cs_type, sdm_type, q_order),
+    line_quad_order_arbitrary_(q_order),
+    tri_quad_order_arbitrary_(q_order),
+    quad_quad_order_arbitrary_(q_order),
+    tet_quad_order_arbitrary_(q_order),
+    hex_quad_order_arbitrary_(q_order),
+    wedge_quad_order_arbitrary_(q_order)
+{
+  line_quad_order_arbitrary_.SetRange({-1.0, 1.0});
+
+  CreateCellMappings();
+}
```

Comment:
Basically, the quadratures can be recycled given the same element type and are fairly simple to create, however, both their storage and initialization can become expensive in numbers. For example, a quadrilateral quadrature has a Legendre quadrature on the inside. Initializing 50,000 legendre quadratures would become noticeable if there are 50,000 quads there. Since they can be recycled and easy to store, we just go ahead and store them.

Reply:
I wasn't proposing initializing on a per-cell mapping basis, but rather, just not initializing hex quadrature in a slab problem (as an example). I suppose for mixed element type meshes, this can become hairy with checks quite quickly, making initializing everything the cleaner alternative.